src/main/java/edu/harvard/iq/dataverse/api/Admin.java

```diff
@@ -1099,6 +1104,91 @@ public Response registerDataFileAll() {
+		Integer successes = 0;
+		Integer alreadyUpdated = 0;
+		Integer rehashed = 0;
+		logger.info("Num = " + num);
+		if (num <= 0)
+			num = Integer.MAX_VALUE;
+		DataFile.ChecksumType cType = null;
+		try {
+			cType = DataFile.ChecksumType.fromString(alg);
+		} catch (IllegalArgumentException iae) {
+			return error(Status.BAD_REQUEST, "Unknown algorithm");
+		}
+		logger.info("Starting to rehash: analyzing " + count + " files. " + new Date());
+		logger.info("Hashes not created with " + alg + " will be verified, and, if valid, replaced with a hash using "
+				+ alg);
+		try {
+			User u = findAuthenticatedUserOrDie();
+			if (!u.isSuperuser())
+				return error(Status.UNAUTHORIZED, "must be superuser");
+		} catch (WrappedResponse e1) {
+			return error(Status.UNAUTHORIZED, "api key required");
+		}
+
+		for (DataFile df : fileService.findAll()) {
+			if (rehashed.intValue() >= num)
+				break;
+			try {
+				if (!df.getChecksumType().equals(cType)) {
+					rehashed++;
+					logger.info(
+							rehashed + ": Datafile: " + df.getFileMetadata().getLabel() + ", " + df.getIdentifier());
+					// verify hash and calc new one to replace it
+					StorageIO<DataFile> storage = df.getStorageIO();
+					storage.open(DataAccessOption.READ_ACCESS);
+					InputStream in = storage.getInputStream();
+					if (in == null)
+						logger.warning("Null stream");
+					String currentChecksum = FileUtil.CalculateChecksum(in, df.getChecksumType());
+					if (currentChecksum.equals(df.getChecksumValue())) {
+						logger.info("Current checksum for datafile: " + df.getFileMetadata().getLabel() + ", "
+								+ df.getIdentifier() + " is valid");
+						storage.open(DataAccessOption.READ_ACCESS);
+						InputStream in2 = storage.getInputStream();
+						if (in2 == null)
+							logger.warning("Null stream");
+						String newChecksum = FileUtil.CalculateChecksum(in2, cType);
+
+						df.setChecksumType(cType);
+						df.setChecksumValue(newChecksum);
+						successes++;
+					} else {
+						logger.warning("Problem: Current checksum for datafile: " + df.getFileMetadata().getLabel()
+								+ ", " + df.getIdentifier() + " is INVALID");
+					}
+				} else {
+					alreadyUpdated++;
+					if (alreadyUpdated % 100 == 0) {
+						logger.info(alreadyUpdated + " of  " + count
+								+ " files are already have hashes with the new algorithm. " + new Date());
+					}
+				}
+				if (successes % 100 == 0) {
+					logger.info(successes + " of  " + count + " files rehashed successfully. " + new Date());
+				}
+
+			} catch (Exception e) {
+				logger.info("Unexpected Exception: " + e.getMessage());
+
+			}
+		}
+		logger.info("Final Results:");
+		logger.info(
+				alreadyUpdated + " of  " + count + " files already had hashes with the new algorithm. " + new Date());
+		logger.info(rehashed + " of  " + count + " files to rehash. " + new Date());
+		logger.info(
+				successes + " of  " + rehashed + " files successfully rehashed with the new algorithm. " + new Date());
```

Comment:
This is a lot of logging at the "info" level. Should it be reduced to "fine"?

Reply:
Actually, I was going to suggest that we keep the more verbose logging. This is a very special operation; nobody's going to be changing the checksum algorithms on their files every day. And it's something important enough, so could be a good thing to have some "paper trail" of this switch having been performed.

Reply:
Up to you guys. As is now, the summary is info level and there are warnings for any file that doesn't match (or can't be read, etc.). The original hash is deleted in this operation though, so if keeping the original one documented is trail-worthy, a per-file info message could be added with the original algorithm and hash and file id (storageidentifier?).